sklearn/tests/test_grid_search.py

```diff
@@ -92,6 +92,10 @@ def score(self):
 y = np.array([1, 1, 2, 2])
 
 
+def assert_grid_iter_equals_getitem(grid):
+    assert_equal(list(grid), [grid[i] for i in range(len(grid))])
```

Comment:
I'm not sure what ``list(grid)`` does now? Is ``__getitem__`` lower priority than whatever other mechanism is used to convert generators?